libraries/chain/include/eosio/chain/block.hpp

```diff
@@ -93,9 +93,9 @@ namespace eosio { namespace chain {
    public:
       signed_block() = default;
       explicit signed_block( const signed_block_header& h ):signed_block_header(h){}
-      signed_block( signed_block&& ) = default;
+      signed_block( signed_block&& ) = default;                // can be =delete once http push_block removed
```

Comment:
Why delete the move constructor? If you don't get a rvalue, meaning something that is not referenced anywhere, the problem is in the caller.

Reply:
There should be no use-case to move a block. We use shared_ptr to a block everywhere.

Reply:
If there is no reason to move a block, then `read_write::push_block(read_write::push_block_params&& params, ...)` should not receive a rvalue I think.

Reply:
It does that to match what the macro code used in `chain_api_plugin` expects for all of them.